frontend/src/store/modules/shootStaging.js

```diff
@@ -168,19 +168,20 @@ const actions = {
     commit('setDns', {
       domain,
       providers,
-      primaryProviderId
+      primaryProviderId,
+      noCustomDomain
     })
   }
 }
 
 // mutations
 const mutations = {
-  setDns (state, { domain, providers = [], primaryProviderId = null }) {
+  setDns (state, { domain, providers = [], primaryProviderId = null, noCustomDomain = false }) {
     state.dnsDomain = domain
     state.dnsProviders = keyBy(providers, 'id')
     state.dnsProviderIds = map(providers, 'id')
     state.dnsPrimaryProviderId = primaryProviderId
-    state.dnsPrimaryProviderValid = !state.dnsDomain || !!state.dnsPrimaryProviderId
+    state.dnsPrimaryProviderValid = !state.dnsDomain || !!state.dnsPrimaryProviderId || noCustomDomain
```

Comment:
since `!primaryProviderId === noCustomDomain` this means `!!state.dnsPrimaryProviderId || noCustomDomain` is always true?

Reply:
I would add a property:
```js
state.dnsPrimaryProviderRequired =  !!primaryProviderId
```
but do not quite understand the reasoning behind it

Reply:
This is not the case. This statement is only true for _already created shoots_. This is the whole purpose of my change. For created shoots without a custom domain, the domain field is populated by the custom domain, while _dnsPrimaryProviderId_  undefined. However, the information if the shoot is already created is not avaialble in the store (or am i wrong?). Of course, we could also store this information but I thought it would be easier to understand if we provide the actual information in a central place (mixin).
Same applies for your comment above:
>I absolutely do not understand why we need to introduce this component parameter. I though the goal of my last my last PR was to get rid of these kind of component parameters. All the necessary information is already available on the store.

We _could_ get the information by evaluating both the `dnsPrimaryProviderId` store state and the `created` prop but I wanted to keep the logic in a central place. I did not put it inside the store, as we also need the computed prop for the details card which has nothing to do with the store... ? 
Maybe we need to discuss this online